osu.Game/Overlays/Comments/CommentEditor.cs

```diff
@@ -37,6 +42,13 @@ public abstract partial class CommentEditor : CompositeDrawable
 
         protected TextBox TextBox { get; private set; } = null!;
 
+        protected readonly IBindable<APIUser> User = new Bindable<APIUser>();
+
+        [Resolved]
+        private IAPIProvider api { get; set; } = null!;
+
+        private LocalisableString placeholderText => api.IsLoggedIn ? TextBoxPlaceholder : textBoxPlaceholderLoggedOut;
```

Comment:
I would probably make this a `getPlaceholderText(bool loggedIn)` thing or something. A property like this hiding conditional logic inside is pretty insidious and easy to miss - doing stuff like `PlaceholderText = placeholderText` looks much to innocuous - if not straight up redundant - if you don't realise that it's not a pure data accessor and in fact depends on other data.

Probably would also make it `protected`, and make it supersede `TextBoxPlaceholder` entirely. Yes the offline branch will be a bit redundant but also a bit more explicit as to what is what and that the comment box may be unavailable sometimes.